src/recommendations/data/hooks/useAlgoliaRecommendations.jsx

```diff
@@ -0,0 +1,85 @@
+import { useEffect, useState } from 'react';
+
+import algoliasearchHelper from 'algoliasearch-helper';
+
+import {
+  getLocationRestrictionFilter,
+  getPrecisionRadius,
+  initializeSearchClient,
+} from '../../../data/algolia';
+import isOneTrustFunctionalCookieEnabled from '../../../data/utils/oneTrust';
+import processCourseSearchResult from '../algoliaResultsParser';
+import {
+  LEVEL_FACET, MAX_RECOMMENDATIONS, PRODUCT_FACET, PRODUCT_TYPE_COURSE,
+} from '../constants';
+
+const INDEX_NAME = process.env.ALGOLIA_AUTHN_RECOMMENDATIONS_INDEX;
+
+/**
+ * This hooks returns Algolia recommendations only if functional cookies are enabled. * @param userCountry
+ * @param educationLevel
+ * @param shouldFetch
+ * @returns {{isLoading: boolean, recommendations: *[]}}
+ */
+const useAlgoliaRecommendations = (userCountry, educationLevel) => {
+  const functionalCookiesEnabled = isOneTrustFunctionalCookieEnabled();
+  const [recommendations, setRecommendations] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    if (!functionalCookiesEnabled) {
+      setIsLoading(false);
+      return;
+    }
+
+    const getSearchFiltersQueryString = () => {
+      if (userCountry) {
```

Comment:
Please move this condition inside `getLocationRestrictionFilter` function.

Reply:
IMO opinion it doesn't make sense to move conditions inside the function. The purpose of the function is to get the country code and prepare a restriction filter. It should be our responsibility to make sure to only call this function if we have a country code. 
Second, if we move the condition inside the function, there will be no visibility in the hook that this filter can either exist or not. Hook's implementation will give an impression that country restriction is always getting appended.

What do you suggest?

Reply:
I think it should be the function's responsibility to handle exceptions too.